include/s3select.h

```diff
@@ -1806,10 +1813,38 @@ class csv_object : public base_s3object
 
 public:
 
+  void result_values_to_string(multi_values& projections_resuls, std::string& result)
+  {
+    size_t i = 0;
+    for(auto res : projections_resuls.values)
+    {
+            result.append( res->to_string() );
+
+            if (m_csv_defintion.quote_fields_always) {
+              std::ostringstream quoted_result;
+              quoted_result << std::quoted(result);
```

Comment:
the std::quoted works on *final* result string which is appended with all projections.

meaning, it quotes what is already quoted.

each projection should be quoted separately, and that should be appended to the final result.